### ISSUE TITLE
Try using other network architectures

### ISSUE THREAD

**Issue by marco-c** [marco-c/autowebcompat#1]

**Comment by Zzoey** [comment]
@marco-c  I am planning to add, Resnet along with a siamese module as class now. Later I will proceed using pretrained models of these networks (Vgg, Resnet) and compare the accuracies. Then we will have a better picture about how to proceed (As I said in #2).  If you have something specific in mind, please let me know.

**Comment by nok** [comment]
We can test pre-defined and later fine-tuned networks, but I'm concerned about the size of the input layer, because a small length could distort and skew the relevant visual representation of the bug.

**Comment by Zzoey** [comment]
@nok Yes, I understand. 
To detect differences, Y+D and N in a better way or even Y and D+N finding attention based ROIs and feeding those patches to the NN would be a good idea. Because, some times the advertisements act as blocks and also grab most of the visual representation. Although this can be considered as an add-on to improve accuracy after using pre-trained and fine tuned networks. I guess, this method can also act as a solution to the case you mentioned.

**Comment by ManasviKundalia** [comment]
@marco-c Do you have any specific architecture in mind? Or perhaps we could add support for attention based layers ?

**Comment by hardik124** [comment]
@marco-c Do you think a YOLO type network will be good to identify relevant ROIs and then classify them? 
I am new to this repository, so please correct me if I am wrong.

**Comment by marco-c** [comment]
> @marco-c Do you have any specific architecture in mind? Or perhaps we could add support for attention based layers ?

> @marco-c Do you think a YOLO type network will be good to identify relevant ROIs and then classify them?
I am new to this repository, so please correct me if I am wrong.

I don't have specific ideas, we should just add whatever we think might work for us and then compare the results.

**Comment by hardik124** [comment]
@marco-c , Is there a documentation I can read? 
I have very little knowledge of this repository ,but if we target some rendering glitches or bugs, we can use YOLO type network.

**Comment by marco-c** [comment]
> @marco-c , Is there a documentation I can read?

I'm going to improve the documentation (https://github.com/marco-c/autowebcompat/issues/77). At the moment, the documentation is the code (which is not too much anyway).

> I have very little knowledge of this repository ,but if we target some rendering glitches or bugs, we can use YOLO type network.

Sounds good to me. I have never used them myself, but it's worth testing them and comparing them with other solutions.

**Comment by marco-c** [comment]
> @marco-c , Is there a documentation I can read?

I've updated README to add more documentation. Feel free to ask if you still have some doubts.

**Comment by hardik124** [comment]
@marco-c . Thanks.

**Comment by Trion129** [comment]
I found this architecture Simnet by Amazon Development Services, it uses a variation of Siamese network, they use 2 extra shallower models trained on downsampled images alongside the ImageNet, 
The final results are better.
https://arxiv.org/pdf/1709.08761.pdf
![image](https://user-images.githubusercontent.com/8445249/36797217-6baf11dc-1ccd-11e8-884f-1855d69031d5.png)

I can try implementing this, what are your thoughts?

**Comment by marco-c** [comment]
@Trion129 sounds good, please file an issue (`Try using Simnet architecture`) with the same description you put in your comment and add a reference to this issue.

**Comment by Shashi456** [comment]
@marco-c  should we try out vgg19 , the microsoft resnet and other forms of inception later on ? i mean can we add them in networks.py?

**Comment by marco-c** [comment]
Sure, we can add them.

**Comment by marco-c** [comment]
Closing in favor of https://github.com/marco-c/autowebcompat/issues/194.
